Cilsil/Cil/Parsers/ExceptionParser.cs

```diff
@@ -0,0 +1,293 @@
+using Cilsil.Utils;
+using Cilsil.Sil;
+using Cilsil.Sil.Types;
+using Mono.Cecil.Cil;
+using Mono.Cecil;
+using Cilsil.Sil.Instructions;
+using Cilsil.Sil.Expressions;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Cilsil.Cil.Parsers
+{
+    internal class ExceptionParser : InstructionParser
+    {
+        protected override bool ParseCilInstructionInternal(Instruction instruction,
+                                                            ProgramState state)
+        {
+            var endBlockOffset = state.ExceptionBlockStartToEndOffsets[state.CurrentInstruction.Offset];
+            
+            /*node1 preds: succs:2 3 exn: Instructions
+            n$47=*&amp;return:void [line 55];   # load return
+            *&amp;return:void=null [line 55];
+            n$48=_fun___unwrap_exception(n$47:void) [line 55];*/
+
+            // Create exception handler node.
+            var exceptionHandlerNode = new StatementNode(state.CurrentLocation,
+                                                         StatementNode.StatementNodeKind.ExceptionHandler,
+                                                         state.ProcDesc);
+            // Load expression from stack. 
+            (var expression, var expressionType) = state.Pop(); 
+            var identifier = state.GetIdentifier(Identifier.IdentKind.Normal);
+            exceptionHandlerNode.Instructions.Add(
+                    new Load(identifier,
+                             expression,
+                             expressionType,
+                             state.CurrentLocation));
+
+            // Clear the loaded expression to null.
+            var exceptionOwnerExpression = new VarExpression(identifier);
+            var storeValueIntoVariable = new Store(expression,
+                                                   new ConstExpression(new IntRepresentation(0, false, true)),
+                                                   expressionType,
+                                                   state.CurrentLocation);
+            exceptionHandlerNode.Instructions.Add(storeValueIntoVariable);
+            
+            // n${i}=_fun___unwrap_exception(n${i-1}:last var type*)
+            CreateExceptionCall(state,
+                                exceptionOwnerExpression,
+                                expressionType,
+                                out var exceptionType,
+                                out var returnVariable,
+                                out _,
+                                out var exceptionCall);
+            exceptionHandlerNode.Instructions.Add(exceptionCall);
+            RegisterNode(state, exceptionHandlerNode, true);
+
+            Expression catchVariable = null;
+            if (exceptionType != state.Method.Module.TypeSystem.Object)
```

Comment:
A good comment here would be to describe what this expression is about/what kinds of places we see it?

Reply:
You should say some comment "case for handling generic, i.e. any exception type" vs "case for handling specified type of exception, i.e. IOException"

Reply:
revise -- comment that finally blocks have "Object" exceptionType (though you should assign null), and catch block will have some other exception type, and that thus in this area you are specifically handling the catch block case.

Reply:
Done